### PR TITLE
C++: Exclude unevaluated accesses in `BufferAccess`

### PR DESCRIPTION
**Pull request by MathiasVP**
Fixes https://github.com/github/codeql/issues/12576.

### CODE REVIEW

File: cpp/ql/test/query-tests/Security/CWE/CWE-119/semmle/tests/OverflowStatic.expected [hunk 1]
```diff
@@ -5,4 +5,5 @@
 | tests.cpp:245:42:245:42 | 6 | Potential buffer-overflow: 'global_array_5' has size 5 not 6. |
 | tests.cpp:349:2:349:14 | access to array | Potential buffer-overflow: 'charArray' has size 10 but 'charArray[10]' may be accessed here. |
 | tests.cpp:350:17:350:29 | access to array | Potential buffer-overflow: 'charArray' has size 10 but 'charArray[10]' may be accessed here. |
+| tests.cpp:608:33:608:43 | access to array | Potential buffer-overflow: 'buffer' has size 100 but 'buffer[101]' may be accessed here. |
```
Comment (jketema) [inline cpp/ql/test/query-tests/Security/CWE/CWE-119/semmle/tests/OverflowStatic.expected:8]: Is this still expected?

Comment (MathiasVP) [reply cpp/ql/test/query-tests/Security/CWE/CWE-119/semmle/tests/OverflowStatic.expected:8]: Hmmm, wait. No, it's not 🤔. Looks like I was too fast there. Let me look into it!

Comment (MathiasVP) [reply cpp/ql/test/query-tests/Security/CWE/CWE-119/semmle/tests/OverflowStatic.expected]: Fixed in https://github.com/github/codeql/pull/12611/commits/2ce0d2b7ee6482209dbd07d9f1bf8f9da1170e26. I had just forgotten to accept test changes in that file 😂.

**Review by jketema — APPROVED** [review]
LGTM if DCA is happy.

**Comment by MathiasVP** [comment]
DCA looks fine

**Pull request merged**